### PR TITLE
feat(globe): size the globe consistently across aspect ratios

### LINKED ISSUE #64: Globe aspect-aware sizing (cameraForViewport)
## Parent

#61

## What to build

Make the hero globe read at a consistent size across aspect ratios. Replace the static camera with an aspect-aware fit so the sphere fills roughly 85% of the *shorter* viewport dimension and is never cropped, re-fitting on viewport resize. This fixes the small-globe / large-side-margin look on landscape phones and md (~768-1024px) widths. There is one root cause (the PerspectiveCamera fits its FOV vertically), so portrait, landscape, and md are corrected together.

## Acceptance criteria

- [x] A pure `cameraForViewport(aspect)` derives the camera FOV from the viewport aspect (distance held at today's value)
- [x] Wide/short and md viewports: the globe fills ~85% of the shorter (height) dimension, never cropped
- [x] Portrait/tall viewports: framing unchanged from today (no regression; the globe never gets smaller than now)
- [x] The fit updates on viewport resize
- [x] Unit tests: wide-and-short narrows the FOV (bigger globe); tall is clamped to today's FOV; the ~85% fill holds across the wide aspect range
- [x] build + lint + typecheck + test green

## Approach (decided 2026-06-05)

Approach C: enlarge wide/short and md viewports only; clamp tall viewports to today's framing so the portrait hero does not shrink. Implemented by **narrowing FOV** (camera distance held at 3.5), which leaves the camera-arc and OrbitControls untouched. The literal "85% on every orientation" rule was dropped: in portrait the shorter side is the narrow width, so it would shrink the globe (~630px to ~330px) on a screen that already looks right.

## Blocked by

None - can start immediately.

### PR DESCRIPTION
**Pull request by taewanu**
Closes #64

## What

The hero globe looked small with large side margins on landscape phones and md (~768-1024px) widths: a `PerspectiveCamera` fits its FOV vertically, so the globe tracked viewport height regardless of width. This makes the camera FOV aspect-aware.

- New pure `cameraForViewport(aspect)` (`camera-fit.ts`): narrows the FOV on wide/short viewports so the globe fills ~85% of the shorter (height) dimension, and clamps tall viewports to the base 45 degree FOV. Distance is held at 3.5, so the camera-arc and OrbitControls are untouched.
- `AspectCameraFit` leaf in `globe-scene.tsx`: subscribes to the canvas width/height and sets `camera.fov` on resize via R3F's imperative `get()` accessor (touches only FOV, never position).
- 9 unit tests for the pure function.

## Approach (C, decided in grill)

"85% of the shorter side on every orientation" was dropped: in portrait the shorter side is the narrow width, which would shrink the globe (about 630px to 330px) on a screen that already looks right. So the rule is enlarge wide/short and md only; tall/portrait keeps today's framing via the FOV clamp. AC reworded on the issue accordingly.

## Verification

Local matrix green: typecheck, lint, test (188, +9), build. Visually checked in headless Chrome at portrait (393x852), landscape (852x393), and md (900x700): landscape/md globe now fills ~85% of height (prominent, no longer small-with-margins), portrait unchanged, KR focus + pins + auto-rotate intact, no console errors.

### CODE REVIEW

**Comment by taewanu** [comment]
✅ Reviewed `ab3ef544fe93` — 0 findings

_Scope: `dc06afe29fdf..ab3ef544fe93`_

<details><summary>1 file</summary>

- `src/components/globe/camera-fit.test.ts`

</details>

**Review by taewanu — COMMENTED** [review]
_Generated by pr-review-agent v0.0.0 (preview release). Rendering, voice, anchoring, and failure handling are still landing in slices. See open issues for in-progress work._

---

Clean FOV-fit helper with a held-constant distance, so OrbitControls and the camera arc stay untouched. Math checks out: the test's fill formula is consistent with the implementation, and clamping, distance-invariance, and the FOV ceiling are all pinned.

- Implementation and test math verified consistent
- One guard branch left untested
- Nothing else high-signal to flag

---

🤖 Auto-submitted by [pr-review-agent](https://github.com/taewanu/pr-review-agent). Edit as needed.

File: src/components/globe/camera-fit.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+/**
+ * FOV for a viewport aspect (width / height); distance is held constant so the
+ * camera-arc and OrbitControls are unaffected. A PerspectiveCamera fits its FOV
+ * vertically, so wide/short viewports narrow it to fill TARGET_FILL of the height;
+ * tall viewports clamp to the base FOV so the portrait globe never shrinks.
+ */
+export function cameraForViewport(aspect: number): CameraFit {
+  if (!(aspect > 0)) return { fov: BASE_FOV_DEG, distance: BASE_DISTANCE };
```
Comment (taewanu) [inline src/components/globe/camera-fit.ts:22]: _🐛 bug_ | _🟡 nit_

**Add a case for the `!(aspect > 0)` guard.** The suite covers the normal aspect range but never the branch that exists because `width / height` produces `0`, `NaN`, or `Infinity` before layout settles.

- `AspectCameraFit` feeds `width / height` straight in, and a zero-height container yields `NaN`
- The guard returning base FOV is the only behavior keeping that from corrupting the projection matrix
- A `cameraForViewport(0)` / `NaN` assertion locks it

---

🤖 _AI-drafted_

Comment (taewanu) [reply src/components/globe/camera-fit.ts:22]: Addressed in `ab3ef54` — added tests for the guard branch: `0` / negative / `NaN` return the base FOV, and a degenerate `Infinity` aspect stays finite (never crashes). Test count 188 → 190.

Comment (taewanu) [reply src/components/globe/camera-fit.ts:22]: **Confirmed at `src/components/globe/camera-fit.test.ts` L48-L63.** The guard now has assertions for `0` / `-1` / `NaN` returning base FOV, plus an `Infinity` case that stays finite under the base FOV.

**Review by taewanu — COMMENTED** [review]
_Generated by pr-review-agent v0.0.0 (preview release). Rendering, voice, anchoring, and failure handling are still landing in slices. See open issues for in-progress work._

---

Two edge-case tests pin the `cameraForViewport` aspect guard.

- Assertions match the implementation: non-positive and NaN return the base, Infinity stays finite via `Math.min(1, aspect)`
- Test-only change, no production code touched
- Nothing high-signal to flag

---

🤖 Auto-submitted by [pr-review-agent](https://github.com/taewanu/pr-review-agent). Edit as needed.
